src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -51,6 +56,47 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
         
     }
 
+    @Override
+    public ChallengeDTO updateChallenge(ChallengeDTO challengeDTO) throws NotFoundException {
+
+        Long userId;
+
+        if(challengeDTO.getCreator().getId() == null)
+        {
+            throw new InvalidFieldException().setMessage("Un challenge trebuie sa aiba un creator");
+        }
+
+        Long challengeCategoryId = challengeDTO.getChallengeCategory().getId();
+
+        Optional<Long> challengeCategoryOptional = Optional.ofNullable(challengeCategoryId);
+
+        challengeCategoryOptional.ifPresent( id -> challengeCategoryRepository.findById(id).
+                        orElseThrow(()-> new InvalidFieldException().setMessage("Categoria aleasa trebuie sa existe"))
+                    );
+
```

Comment:
Poti sa reduci numarul de linii concatenand ce ai scris la linia 69 cu ce este la 71.  De asemenea, if-ul de mai jos if(!challengeCategoryOptional.isPresent()) nu mai are rost, ca tu pe ifPresent,la linia 74, arunci exceptie, ceea ce inseamna ca codul tau nu o sa se mai execute mai departe.

Reply:
La 74 verific daca id-ul este prezent dar nu corespunde unui ChallengeCategory din baza de date si arunc exceptie. La 77 verific daca challengeCategoryDTO are id-ul "null" si atunci fac tot obiectul ca fiind "null". Am schimbat ordinea lor in urmatorul push in ideea ca e mai intuitiv sa verific daca exista prima data si abia apoi daca e gresit. Am facut si concatenarea. Or sa apara la urmatorul push.